src/parquet/parquet.h

```diff
@@ -117,6 +117,22 @@ class InMemoryInputStream : public InputStream {
   int64_t offset_;
 };
 
+// A wrapper for InMemoryInputStream to manage the memory.
+class ScopedInMemoryInputStream : public InputStream {
+ public:
+  ScopedInMemoryInputStream(const int64_t len);
+  ~ScopedInMemoryInputStream();
+  uint8_t* data();
+  int64_t size();
+  virtual const uint8_t* Peek(int num_to_peek, int* num_bytes);
+  virtual const uint8_t* Read(int num_to_read, int* num_bytes);
+
+ private:
+  int64_t len_;
+  uint8_t* buffer_;
```

Comment:
malloc is perfectly fine in this case. std::vector will call it, internally, too; there is no need to obscure this memory allocation. And we don't need vector or a smart pointer to manage memory deallocation - this is done in ~ScopedInMemoryInputStream() dctor.